p2p/net/swarm/swarm_dial.go

```diff
@@ -484,6 +491,9 @@ func (s *Swarm) dialAddr(ctx context.Context, p peer.ID, addr ma.Multiaddr) (tra
 
 	start := time.Now()
 	connC, err := tpt.Dial(ctx, addr, p)
+
+	s.bhd.RecordResult(addr, err == nil)
```

Comment:
```suggestion
	// We're recording any error as a failure here.
	// Notably, this also applies to cancelations (i.e. if another dial attempt was faster).
	// This is ok since the black hole detector uses a very low threshold (5%).
	s.bhd.RecordResult(addr, err == nil)
```